src/js/crypto.js

```diff
@@ -13,7 +13,65 @@
  * limitations under the License.
  */
 
-var hashes = ['sha1'];
+var IOTJS_SHA1 = 4;
+var IOTJS_SHA256 = 6;
+
+var hashes = ['sha1', 'sha256'];
+
+function Verify(signtype) {
+  if (!(this instanceof Verify)) {
+    return new Verify(signtype);
+  }
+
+  signtype = signtype.toLowerCase();
+
+  if (hashes.indexOf(signtype) < 0) {
+    throw new Error('Unknown signing algorithm.' +
+                    'Please use crypto.getSignatures()');
+  }
+
+  Object.defineProperty(this, 'hashtype', {
+    value: signtype,
+    writable: false,
+    enumerable: true,
+  });
+}
+
+Verify.prototype.update = function(data) {
+  if (this.data) {
+    if (Buffer.isBuffer(data)) {
+      this.data = Buffer.concat([this.data, data]);
+      return;
+    }
+
+    this.data = Buffer.concat([this.data, new Buffer(data)]);
+    return;
+  }
+
+  if (Buffer.isBuffer(data)) {
+    this.data = data;
+  }
```

Comment:
You need a `return` or `else` statement.

Reply:
Oh right, thank you!